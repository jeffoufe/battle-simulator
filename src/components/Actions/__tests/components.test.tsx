import React from 'react';
import { shallow } from 'enzyme';

import {
    ActionsContainer,
    ActionsInfo,
    GameOver,
    Victory,
    AttackButton
} from '../components'

describe('Actions Components', () => {
    it('ActionsContainer', () => {
        const wrapper = shallow(
            <ActionsContainer>ActionsContainer</ActionsContainer>
        );
        expect(wrapper).toMatchSnapshot();
    })

    it('ActionsInfo', () => {
        const wrapper = shallow(
            <ActionsInfo>ActionsInfo</ActionsInfo>
        );
        expect(wrapper).toMatchSnapshot();
    })

    it('GameOver', () => {
        const wrapper = shallow(
            <GameOver>GameOver</GameOver>
        );
        expect(wrapper).toMatchSnapshot();
    })

    it('Victory', () => {
        const wrapper = shallow(
            <Victory>GameOver</Victory>
        );
        expect(wrapper).toMatchSnapshot();
    })

    describe('AttackButton', () => {
        const testFn = jest.fn();
        const wrapper = shallow(
            <AttackButton onClick={testFn}>
                AttackButton
            </AttackButton>
        );

        it('Snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })

        it('OnClick', () => {
            wrapper.find('button').simulate('click');
            expect(testFn).toHaveBeenCalled();
        })
    })
})