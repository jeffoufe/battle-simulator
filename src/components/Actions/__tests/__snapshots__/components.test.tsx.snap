// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Components ActionsContainer 1`] = `
<div
  className="sc-AxjAm dRhhVH"
>
  ActionsContainer
</div>
`;

exports[`Actions Components ActionsInfo 1`] = `
<p
  className="sc-AxirZ dEBbzp"
>
  ActionsInfo
</p>
`;

exports[`Actions Components AttackButton Snapshot 1`] = `
<button
  className="sc-AxhUy fOyfqh"
  onClick={[MockFunction]}
>
  AttackButton
</button>
`;

exports[`Actions Components GameOver 1`] = `
<p
  className="sc-AxirZ sc-AxiKw jcmCxV"
>
  GameOver
</p>
`;

exports[`Actions Components Victory 1`] = `
<p
  className="sc-AxirZ sc-AxhCb ewOrJJ"
>
  GameOver
</p>
`;
